controllers/node_label_module_version_reconciler.go

```diff
@@ -0,0 +1,257 @@
+				labelsPerModule[mapKey].moduleLoaderVersionLabel = value
+			}
+			if utils.IsDevicePluginVersionLabel(key) {
+				labelsPerModule[mapKey].devicePluginVersionLabel = value
+			}
+		}
+	}
+
+	return labelsPerModule
+}
+
+func (nlmvha *nodeLabelModuleVersionHelper) getModuleLoaderPods(ctx context.Context, nodeName string) ([]v1.Pod, error) {
+	var modulePodsList v1.PodList
+	fieldSelector := client.MatchingFields{"spec.nodeName": nodeName}
+	labelSelector := client.MatchingLabels{constants.DaemonSetRole: constants.ModuleLoaderRoleLabelValue}
+	err := nlmvha.client.List(ctx, &modulePodsList, labelSelector, fieldSelector)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get list of all module loader pods for on node %s: %v", nodeName, err)
+	}
+	return modulePodsList.Items, nil
+}
+
+func (nlmvha *nodeLabelModuleVersionHelper) updateNodeLabels(ctx context.Context, nodeName string, reconLabelsRes *reconcileLabelsResult) error {
+	node := v1.Node{}
+
+	if err := nlmvha.client.Get(ctx, types.NamespacedName{Name: nodeName}, &node); err != nil {
+		return fmt.Errorf("could not get node %s: %v", nodeName, err)
+	}
+
+	nodeCopy := node.DeepCopy()
+
+	if node.Labels == nil && len(reconLabelsRes.labelsToAdd) > 0 {
+		node.Labels = make(map[string]string, 1)
+	}
+
+	for label, labelValue := range reconLabelsRes.labelsToAdd {
+		node.Labels[label] = labelValue
+	}
+	for _, label := range reconLabelsRes.labelsToDelete {
+		delete(node.Labels, label)
+	}
+
+	return nlmvha.client.Patch(ctx, &node, client.MergeFrom(nodeCopy))
+}
+
+func (nlmvha *nodeLabelModuleVersionHelper) reconcileLabels(modulesLabels map[string]*modulesVersionLabels,
+	moduleLoaderPods []v1.Pod) *reconcileLabelsResult {
+
+	reconRes := reconcileLabelsResult{
+		labelsToAdd:    map[string]string{},
+		labelsToDelete: []string{},
+	}
+	for _, moduleLabels := range modulesLabels {
+		label, labelValue, action := getLabelAndAction(moduleLabels)
+		switch action {
+		case deleteAction:
+			reconRes.labelsToDelete = append(reconRes.labelsToDelete, label)
+		case addAction:
+			requeue := verifyModuleLoaderAddAction(moduleLabels.name, moduleLabels.namespace, moduleLoaderPods)
+			if !requeue {
+				reconRes.labelsToAdd[label] = labelValue
+			} else {
+				reconRes.requeue = true
+			}
+		}
+	}
+
+	return &reconRes
+}
+
+func verifyModuleLoaderAddAction(name, namespace string, moduleLoaderPods []v1.Pod) bool {
+	for _, pod := range moduleLoaderPods {
+		podLabels := pod.GetLabels()
+		if pod.Namespace == namespace && podLabels[constants.ModuleNameLabel] == name {
+			return true
+		}
+	}
+	return false
+}
+
+// returns the label, value of the label and action to execute on label (add or delete)
+func getLabelAndAction(moduleLabels *modulesVersionLabels) (string, string, string) {
+	labelActionKey := getModuleVersionLabelsState(moduleLabels)
+	labelAction, ok := labelActionTable[labelActionKey]
+	if !ok || labelAction.action == noneAction {
+		return "", "", noneAction
+	}
```

Comment:
```suggestion
	if !ok {
		return "", "", noneAction
	}
```

The second condition is covered in the switch-case.